### ISSUE TITLE
upgrading to rc2 on infomrix 12.10 FC8 : ERROR - Method not supported : IfxSqliConnect.getSchema()

### ISSUE THREAD

**Issue by begooden-it** [schemaspy/schemaspy#219]
Hi,

I just upgraded to 6.0.0.0 rc2
I am using Informix 12.10FC8 + JDBC Informix 4.40.JC9
I now have the following error
ERROR - Method not supported : IfxSqliConnect.getSchema()

## Expected Behavior
I never had this error with rc1
I should not have this error in rc2

## Current Behavior
I launch schemaSpy the same way I did with rc1, but now have this error

## Possible Solution
Apparently, according to this document, fxSqliConnect.getSchema is not a supported method
[https://www.ibm.com/support/knowledgecenter/en/SSEPGG_9.7.0/com.ibm.db2.luw.apdv.java.doc/src/tpc/imjcc_rjvjdapi.html]

## Steps to Reproduce (for bugs)
0. install schemaSpy 6.0.0.0 rc2
1. configure and run Informix 12.10 xC8
2. Install IBM JDBC 4.10 JC9 64bits
3. launch schemaSpy against an existing database in Informix
4. get this error

## Context
Nothing works

## Your Environment

* Version used: 6.0.0.0 rc2
* Environment name and version Chrome Version 63.0.3239.132 (Build officiel) (64 bits)
* Operating System and version Windows 7
* Java version: V81 build 121

**Comment by npetzall** [comment]
What command line arguments did you use?

I had the same issue adding `-s informix` should solve the problem.

Don't close I want to see if we can make it smarter.

**Comment by begooden-it** [comment]
Hi Nils,





I just logged the SR request before seeing your answer.

Yes Adding the –s Informix solves the issue

So this seems to be the solution



Regarding the other issue, I still reproduce with rc2



Did you test with my DB schema ( rfe ? )

This is the first time I had this issue, I mean with this DB









*De :* Nils Petzäll [mailto:notifications@github.com]
*Envoyé :* dimanche 14 janvier 2018 12:35
*À :* schemaspy/schemaspy <schemaspy@noreply.github.com>
*Cc :* begooden-it <eric.vercelletto@begooden-it.com>; Author <
author@noreply.github.com>
*Objet :* Re: [schemaspy/schemaspy] upgrading to rc2 on infomrix 12.10 FC8
: ERROR - Method not supported : IfxSqliConnect.getSchema() (#219)



What command line arguments did you use?

I had the same issue adding -s informix should solve the problem.

Don't close I want to see if we can make it smarter.

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/schemaspy/schemaspy/issues/219#issuecomment-357505367>,
or mute the thread

**Comment by npetzall** [comment]
This is more or less know issue in #117.

TL;DR #204 
Since different database vendors calls things differently and might have them in a strange order.

JDBC has Catalog -> Schema -> Table since this is located in DatabaseMetaData you might consider it
Database -> Catalog -> Schema -> Table

So specifying Database only should include all catalogs, all schemas, all Tables.
Specifying catalog would narrow it down, schema even more.

DatabaseMetaData.getTables()
catalog - a catalog name; must match the catalog name as it is stored in the database; "" retrieves those without a catalog; null means that the catalog name should not be used to narrow the search
schemaPattern - a schema name pattern; must match the schema name as it is stored in the database; "" retrieves those without a schema; null means that the schema name should not be used to narrow the search
tableNamePattern - a table name pattern; must match the table name as it is stored in the database

In both the catalog case and the schema case they are allow to be null which I assume would include all even those without catalog or schema (since it doesn't narrow the search), "" would give you tables without catalog and/or schema, "%" would give all with catalog and/or schema.

If we created catalog and schema after the fact. We would avoid #160 but this is more or less the proposal in #204 since without catalog and schema in the first place we need to assume that we will get tables from multiple catalogs and schemas.

end of rant.

**Comment by brian-hughes** [comment]
One thing I noticed is that the Informix JDBC tends to like to throw exceptions when it does not support something, even if the JDBC spec says otherwise.  So for set/getSchema, Informix really doesn't support schemas like that, but instead of returning "null" as the API calls for, it throws an exception.  

If it would help I can fix that so it behaves more like a no-op for the setSchema and returns null for getSchema instead of throwing an Exception.  However, that will be for the next release of the driver which is probably a month or two away at least (Informix release cycles are still pretty slow)

**Comment by begooden-it** [comment]
Hi Brian,





There is no real hurry for that. I’ll ask Shawn and Karen what is the next
fix schedule 😊



Say hi to lenexa Team 😊



Eric Vercelletto
Data Management Architect and Owner / Begooden IT Consulting
Board of Directors, International Informix Users group
IBM Champion 2013,2014,2015,2016,2017,2018,2019
[image: ibm-champion-rgb-130px]
Tel:     +33(0) 298 51 3210
Mob : +33(0)626 52 50 68
skype: begooden-it
Google Hangout: eric.vercelletto@begooden-it.com
Email: eric.vercelletto@begooden-it.com
www : http://www.vercelletto.com
www  https://kandooerp.org

[image: cid:image001.jpg@01CDC3E9.1425CBB0]

[image: cid:image002.jpg@01CDC3E9.1425CBB0]

[image: cid:image003.jpg@01CDC3E9.1425CBB0]









*De :* Brian Hughes <notifications@github.com>
*Envoyé :* mercredi 3 avril 2019 16:33
*À :* schemaspy/schemaspy <schemaspy@noreply.github.com>
*Cc :* begooden-it <eric.vercelletto@begooden-it.com>; Author <
author@noreply.github.com>
*Objet :* Re: [schemaspy/schemaspy] upgrading to rc2 on infomrix 12.10 FC8
: ERROR - Method not supported : IfxSqliConnect.getSchema() (#219)



One thing I noticed is that the Informix JDBC tends to like to throw
exceptions when it does not support something, even if the JDBC spec says
otherwise. So for set/getSchema, Informix really doesn't support schemas
like that, but instead of returning "null" as the API calls for, it throws
an exception.

If it would help I can fix that so it behaves more like a no-op for the
setSchema and returns null for getSchema instead of throwing an Exception.
However, that will be for the next release of the driver which is probably
a month or two away at least (Informix release cycles are still pretty slow)

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/schemaspy/schemaspy/issues/219#issuecomment-479515906>,
or mute the thread
